shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java

```diff
@@ -121,51 +96,34 @@ public UpdateControl<ManagedBridge> reconcile(ManagedBridge managedBridge, Conte
-            }
-            return UpdateControl.updateStatus(managedBridge).rescheduleAfter(reconcileIntervalMillis);
-        } else if (!status.isConditionTypeTrue(ManagedBridgeStatus.KNATIVE_BROKER_AVAILABLE)) {
-            status.markConditionTrue(ManagedBridgeStatus.KNATIVE_BROKER_AVAILABLE);
+            status.markConditionFalse(DP_KNATIVE_BROKER_READY_NAME);
+            return handleFailure(managedBridge);
+        } else {
+            status.markConditionTrue(DP_KNATIVE_BROKER_READY_NAME);
             LOGGER.info("The Knative Broker Resource for ManagedBridge '{}' in namespace '{}' is ready", managedBridge.getMetadata().getName(), managedBridge.getMetadata().getNamespace());
         }
 
         // Nothing to check for Authorization Policy
         managedBridgeService.fetchOrCreateBridgeAuthorizationPolicy(managedBridge, path);
-        if (!status.isConditionTypeTrue(ManagedBridgeStatus.AUTHORISATION_POLICY_AVAILABLE)) {
-            status.markConditionTrue(ManagedBridgeStatus.AUTHORISATION_POLICY_AVAILABLE);
-        }
+        status.markConditionTrue(DP_AUTHORISATION_POLICY_READY_NAME);
 
         NetworkResource networkResource = networkingService.fetchOrCreateBrokerNetworkIngress(managedBridge, secret, managedBridge.getSpec().getDnsConfiguration().getHost(), path);
-
         if (!networkResource.isReady()) {
             LOGGER.info("Ingress networking resource for ManagedBridge with id '{}' in namespace '{}' is not ready.",
                     managedBridge.getMetadata().getName(),
                     managedBridge.getMetadata().getNamespace());
-            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
-                status.markConditionFalse(ConditionTypeConstants.READY);
-            }
-            if (!status.isConditionTypeFalse(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE)) {
-                status.markConditionFalse(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE);
-            }
-            return UpdateControl.updateStatus(managedBridge).rescheduleAfter(reconcileIntervalMillis);
-        } else if (!status.isConditionTypeTrue(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE)) {
-            status.markConditionTrue(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE);
+            status.markConditionFalse(DP_NETWORK_RESOURCE_READY_NAME);
+            return handleFailure(managedBridge);
+        } else {
+            LOGGER.info("Ingress networking resource for ManagedBridge with id '{}' in namespace '{}' is ready.", managedBridge.getMetadata().getName(),
+                    managedBridge.getMetadata().getNamespace());
+            status.markConditionTrue(DP_NETWORK_RESOURCE_READY_NAME);
         }
 
-        LOGGER.info("Ingress networking resource for ManagedBridge with id '{}' in namespace '{}' is ready.", managedBridge.getMetadata().getName(), managedBridge.getMetadata().getNamespace());
-
-        if (!status.isReady()) {
+        if (isBridgeStatusChange(managedBridge)) {
```

Comment:
You are recording the operation was complete if there was no change in the `Condition`'s between subsequent iterations.

What happens if it takes more than 1 iteration of the reconsile loop before `Condition`'s change?

It **might not be a problem** if we only relied on k8s invoking our loop in response to changes to Custom Resources on which we depend with the k8s eventing mechanism however we force re-entry with use of `rescheduleAfter(..)` and hence the possibility of our reconcile loop running multiple times before `Condition`'s change increases.

It might be better to check if ALL conditions are `True` to signal completion vs there has not been a change.

Reply:
@manstis It only ever gets to this point in the code if all dependencies are ready, otherwise the reconcile loop is ended early via a call to the `handleFailure(ManagedBridge managedBridge)` method. So at this point, essentially all conditions on the `ManagedBridge` will have been be set to `true` earlier in the reconcile loop, or they are already `true`.  If the conditions were set to `true` in the reconcile loop, then the `isBridgeStatusChange` method detects that and implicitly marks the operation as complete.

@vaibhavjainwiz I think something like a `isReady()` method on the `ManagedBridgeStatus` might make it clearer so that the logic looks something like this:

```
if (isBridgeStatusChange(managedBridge)) {
  if(status.isReady()) {
    // update metrics
  }
}
```
However, I'm not going to block this PR for that change, so if you agree, please create a JIRA and we can follow up.

Reply:
Create below Jira to incorporate this change
https://issues.redhat.com/browse/MGDOBR-1343